README.adoc

```diff
@@ -0,0 +1,116 @@
+plugins are based on link:https://github.com/kubernetes/test-infra/tree/master/prow[Prow] sub project of
+Kubernetes Test Infrastructure.
+
+=== Prerequisites
+
+==== Tooling
+
+You need to have following packages in place:
+
+* `git`
+* `make`
+* `go` (>= v1.9.4)
+* link:https://glide.sh/[`glide`] for dependency management
+
+Assuming that you have all the link:https://golang.org/doc/install[Golang prerequisites] in place (such as `$GOPATH`), clone the repository first:
+
+[source,bash]
+----
+$ git clone https://github.com/arquillian/ike-prow-plugins $GOPATH/src/github.com/arquillian/ike-prow-plugins
+----
+
+NOTE: Have a look how link:https://github.com/moovweb/gvm[Go Version Manager] can help you simplifying configuration and
+management of different versions of Go.
+
+NOTE: You can also use `setup.sh` which takes care of all of the above
+
+==== GitHub settings [[gh-settings]]
+
+You will need two secrets to be able to integrate with GitHub. The `hmac.token` file should contain the token that
+you give to GitHub for validating webhooks. You can generate it using any reasonable randomness-generator, for example
+link:http://random.org[random.org].
+
+The `oauth.token` is an OAuth2 token that has read and write access to the GitHub account. Generate it from
+link:https://github.com/settings/tokens/new[here].
+
+NOTE: Both of these files are ignored by git (see `.gitignore`) so you can keep them in your repository, as some `make`
+targets rely on them.
+
+=== Local Setup
+
+==== Minishift
+
+You can use Minishift for developing plugins locally.
+Follow link:https://docs.openshift.org/latest/minishift/getting-started/installing.html#installing-instructions[official installation guide] to set it up.
+
+===== Initial setup
+
+Before you start developing new plugins or evolving existing ones you need to deploy set of basic Prow services and configuration.
+
+For this purpose use `oc-init` target which will:
+
+* Create new project named `ike-prow-plugins`
+* Add config maps for Prow configuration (`config.yaml`) and its plugins (`plugins.yaml`)
+* Add required secrets (see <<gh-settings>> for details)
+* Apply `cluster/starter.yaml` with basic Prow infra.
+
+The key piece of this deployment is the `hook` service which listens to GitHub events and dispatches them to relevant plugins.
+
+You will have to create a route for this service. Go to `Applications -> Services -> hook` and click `Create Route`.
+Set `/hook` as a path and click `Create`.
+
+===== Pushing to local Minishift repository
+
+To speed up development cycle (and avoid pushing work-in-progress) images to Docker Hub you can use built-in docker repository available in Minishift.
+`Makefile` and templates are adjusted to use it, so the only thing you have to do is to set up few environment variables.
+
+[source,bash]
+----
+eval $(minishift docker-env)
+docker login -u $(oc whoami) -p $(oc whoami -t) $(minishift openshift registry)
```

Comment:
1. Get to the openshift console
2. Copy login command
3. Login 
4. Run this

Reply:
the login command is OK, but then I get the same error:
```
[mjobanek ~/arquillian/ike-prow-plugins] (master)oc login https://192.168.42.147:8443 --token=abcdefg
Logged into "https://192.168.42.147:8443" as "developer" using the token provided.

You don't have any projects. You can try to create a new project, by running

    oc new-project <projectname>

[mjobanek ~/arquillian/ike-prow-plugins] (master)$ docker login -u $(oc whoami) -p $(oc whoami -t) $(minishift openshift registry)
Error response from daemon: Get http://172.30.1.1:5000/v1/users/: dial tcp 172.30.1.1:5000: getsockopt: connection refused
```

Reply:
one question - shouldn't that be 
`oc login -u $(oc whoami) -p $(oc whoami -t) $(minishift openshift registry)`
instead of 
`docker login -u $(oc whoami) -p $(oc whoami -t) $(minishift openshift registry)`

Reply:
Ah ok, then you have to login as `system` user to have required privileges (that's the only difference I can think of). Any password will work.

> one question - shouldn't that be
oc login -u $(oc whoami) -p $(oc whoami -t) $(minishift openshift registry)
instead of
docker login -u $(oc whoami) -p $(oc whoami -t) $(minishift openshift registry)

No, you have to tell your docker daemon to talk with minishift repo for pushing images.

Reply:
ok, the problem was between my chair and my keyboard - I had an old version of minishift and I started with that one and created the projects in it. Then I upgraded to a newer version, but it wasn't backward compatible obviously, so I got that error (and also some other ones). 
I removed the minishift and did the whole process with the newest one and everything works... 
My bad :-)